etl/steps/data/garden/who/2024-08-07/monkeypox.py

```diff
@@ -0,0 +1,196 @@
+            "date": "date",
+            "total_conf_cases": "total_cases",
+            "total_conf_deaths": "total_deaths",
+            "iso3": "iso_code",
+        }
+    )
+
+
+def clean_date(tb: Table) -> Table:
+    tb["date"] = pd.to_datetime(tb.date).dt.date.astype(str)
+    return tb
+
+
+def clean_values(tb: Table) -> Table:
+    tb = tb.sort_values("date", ascending=False)
+    tb["total_cases"] = tb[["country", "total_cases"]].groupby("country", observed=True).cummin()
+    tb["total_deaths"] = tb[["country", "total_deaths"]].groupby("country", observed=True).cummin()
+    return tb.sort_values(["country", "date"])
+
+
+def explode_dates(tb: Table) -> Table:
+    tb_range = pd.concat(
+        [
+            pd.DataFrame(
+                {
+                    "country": country,
+                    "date": pd.date_range(start=tb.date.min(), end=tb.date.max(), freq="D").astype(str),
+                }
+            )
+            for country in tb.country.unique()
+        ]
+    )
+    country_to_iso = tb[["country", "iso_code"]].drop_duplicates().set_index("country").iso_code
+    tb = pr.merge(tb, Table(tb_range), on=["country", "date"], validate="one_to_one", how="right")
+    tb["iso_code"] = tb.country.map(country_to_iso)
+    tb["report"] = tb.total_cases.notnull() | tb.total_deaths.notnull()
+    return tb
+
+
+def add_world(tb: Table) -> Table:
+    tb[["total_cases", "total_deaths"]] = (
+        tb[["country", "total_cases", "total_deaths"]].groupby("country", observed=True).ffill().fillna(0)
+    )
+    world = (
+        tb[["date", "total_cases", "total_deaths"]]
+        .groupby("date", as_index=False, observed=True)
+        .sum()
+        .assign(country="World", report=True)
+    )
+    world = world[world.date < str(datetime.date.today())]
+    return pr.concat([tb, world])
+
+
+def add_regions(tb: Table) -> Table:
+    ds_regions = paths.load_dataset("regions")
+
+    # Add region for each country
+    for region in REGIONS_TO_ADD:
+        countries_in_region = geo.list_members_of_region(
+            region=region,
+            ds_regions=ds_regions,
+        )
+        tb.loc[tb.country.isin(countries_in_region), "region"] = region
+
+    # Calculate regional aggregates
+    regions = (
+        tb[tb.region.notnull()][["region", "date", "total_cases", "total_deaths", "report"]]
+        .groupby(["region", "date"], as_index=False, observed=True)
+        .agg({"total_cases": "sum", "total_deaths": "sum", "report": "max"})
+        .rename(columns={"region": "country"})
+    )
+    regions = regions[regions.date < str(datetime.date.today())]
+    tb = tb.drop(columns="region")
+
+    # Concatenate with tb
+    return pr.concat([tb, regions])
+
+
+def add_population_and_countries(tb: Table) -> Table:
+    ds_population = paths.load_dataset("population")
+
+    # Use population from 2022. Using population per year results in charts with
+    # cumulative cases per million showing jumps around new year.
```

Comment:
heads up that for COVID, I had the same 'problem'. I decided to use 'daily population', which I estimated by linearly-interpolating the yearly values.

I implemented this function:

https://github.com/owid/etl/blob/5c62946e9451e3f62b3b308aee31644e4af25759/etl/steps/data/garden/covid/latest/shared.py#L94-L113

which then I call like:

https://github.com/owid/etl/blob/5c62946e9451e3f62b3b308aee31644e4af25759/etl/steps/data/garden/covid/latest/sequence.py#L398-L403